dkeycache/App/main.cpp

```diff
@@ -59,8 +59,27 @@ void cleanup()
 
 int main(int argc, char* argv[])
 {
-    (void)argc;
-    (void)argv;
+    // process argv
+    std::string host;
+    uint32_t port;
+    std::string host_key = "-h";
+    std::string port_key = "-p";
+    int host_index = -1;
+    int port_index = -1;
+    for (int i = 0; i < argc; i++) {
+        if (argv[i] == host_key && i + 1 < argc && argv[i + 1][0] != '-') {
+            host_index = i + 1;
+        } else if (argv[i] == port_key && i + 1 < argc && argv[i + 1][0] != '-') {
+            port_index = i + 1;
+        }
+    }
+    if (host_index == -1 || port_index == -1)
+    {
+        printf("\nusage: ehsm-dkeycache -h 127.0.0.1 -p 8888\n\n");
+        return -1;
```

Comment:
you can follow the enroll_app, don't use "-h" or "-p"
just directly run as "ehsm-dkeycache <dkeyserver_url>" and the dkeyserver_url="127.0.0.1:8888"

Reply:
ignore it, seems you need to parse the input to get the IP and port

Reply:
It is fixed.please check again.